GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -154,44 +208,97 @@ public void runImport(File file, String serverBase, Map<String, String> criteria
      *
      * @param sheet
      * @param startRow
+     * @param columnIndexToQuestionId
+     * @param questionIdToQuestionDto
      * @return InstanceData
      */
-    public InstanceData parse(Sheet sheet, int startRow) {
+    public InstanceData parseInstance(Sheet sheet, int startRow,
+            Map<Long, QuestionDto> questionIdToQuestionDto, Map<Long, Long> columnIndexToQuestionId) {
 
         // File layout
         // 0. SurveyedLocaleIdentifier
+        // 7. Repeat
         // 1. SurveyedLocaleDisplayName
         // 2. DeviceIdentifier
         // 3. SurveyInstanceId
         // 4. CollectionDate
         // 5. SubmitterName
         // 6. SurveyalTime
-        // 7. Repeat
+
         // 8 - N. Questions
         // N + 1. Digest
 
         Row baseRow = sheet.getRow(startRow);
-        String surveyedLocaleIdentifier = baseRow.getCell(0).getStringCellValue();
-        String surveyedLocaleDisplayName = baseRow.getCell(1).getStringCellValue();
-        String deviceIdentifier = baseRow.getCell(2).getStringCellValue();
-        String surveyInstanceId = baseRow.getCell(3).getStringCellValue();
-        String collectionDate = baseRow.getCell(4).getStringCellValue();
-        String submitterName = baseRow.getCell(5).getStringCellValue();
-        String surveyalTime = baseRow.getCell(6).getStringCellValue();
-        int firstQuestionColumn = 8;
+        if (baseRow == null) {
+            return null;
+        }
+        String surveyedLocaleIdentifier = ExportUtils.parseCellAsString(baseRow.getCell(0));
+        String surveyedLocaleDisplayName = ExportUtils.parseCellAsString(baseRow.getCell(2));
+        String deviceIdentifier = ExportUtils.parseCellAsString(baseRow.getCell(3));
+        String surveyInstanceId = ExportUtils.parseCellAsString(baseRow.getCell(4));
+        Date collectionDate = ExportUtils.parseDate(
+                ExportUtils.parseCellAsString(baseRow.getCell(5)));
+        String submitterName = ExportUtils.parseCellAsString(baseRow.getCell(6));
+        String surveyalTime = ExportUtils.parseCellAsString(baseRow.getCell(7));
 
```

Comment:
For this section where we identify the first 8 columns, we need to use a similar construct as in the previous implementation in order to be backwards compatible. See https://github.com/akvo/akvo-flow/blob/develop/GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java#L285-L313.  Or alternatively we need to identify the column in question by using the header text.